### DISCUSSION TITLE
(A)GPL 라이센스 호환 여부

### DISCUSSION THREAD

**Discussion by tribela** [orioncactus/pretendard#18] [Q&A] [answered]
좋은 폰트 만들어주셔서 감사합니다.

사용하는 프로젝트에 Pretendard 폰트를 적용해보려 하였지만 단독판매 금지조항과 충돌하여 SIL 라이센스는 GPL(+AGPL)과는 호환이 안 된다고 합니다. 하지만 번들 또는 내장하는 경우에는 허용이 된다는 말도 있는데요, AGPL 라이센스를 사용하는 소프트웨어에서 Pretendard를 번들로 묶어도 되는 것인지 명확하게 알고 싶습니다.

**Comment by hiddenest** [comment]
지나가는 행인입니다. 이전에 Font Awesome도 비슷한 논의가 있어서 해당 링크를 첨부해봅니다.
https://github.com/FortAwesome/Font-Awesome/issues/1124#issuecomment-29507471

**Comment by orioncactus** [comment, marked as answer]
안녕하세요! Pretendard를 아껴주셔서 감사드립니다. GPL이 적용받는 범위가 어디까지인지 파악하다 답변이 늦었습니다. 우선 확인해본 문서는 다음과 같습니다:
- https://github.com/FortAwesome/Font-Awesome/issues/1124#issuecomment-29507471 및 https://scripts.sil.org/cms/scripts/page.php?site_id=nrsi&id=ofl-faq_web#68092c0f
- https://www.gnu.org/licenses/gpl-faq.html
- https://scripts.sil.org/cms/scripts/page.php?item_id=OFL_web
- https://en.wikipedia.org/wiki/GNU_General_Public_License#Libraries

위에 hiddenest님께서 올려주신 [Font Awesome의 Issue](https://github.com/FortAwesome/Font-Awesome/issues/1124#issuecomment-29507471)와 OFL의 판매 금지 조항, GPL의 전염성 조항, OFL과 GPL의 FAQ, 그리고 GPL Wikipedia 등에서 봤을 때 정리한 사항은 이렇습니다. 해석에 도움을 주신 [ied206](https://github.com/ied206)님과 [leejh10003](https://github.com/leejh10003)님께 진심으로 감사드립니다.
- GPL로 배포되는 소스를 사용하는 소프트웨어는 자동적으로 GPL이 적용된다.
- GPL의 전염성이 적용되는 대상은 소스 코드를 통해 배포되는 라이브러리 또는 프로그램이다.
- OFL에서는 글꼴 자체를 판매하는 행위를 금지한다. 그러나 GPL에서는 소스 코드 판매가 가능하기 때문에, 이와 대치되어 OFL은 GPL과 호환되지 않는다.
- GPL 소프트웨어에서 별도의 작업으로 간주되는 경우 GPL에 적용되지 않는다.
- GPL의 적용 여부는 GPL로 배포하는 소프트웨어에서 해당 소스가 서로 어떠한 방법으로 작용하는지에 따라 나뉘어진다. 곧 결합된 부분의 변화가 원본의 전체 기능성에 영향을 미치는지를 판단하며, 해당 소스가 독립적이며 별도로 기능하는지, 또는 통합적이며 소스로부터 파생되어 기능하는지에 따라 결정된다.
- 통상적으로 글꼴은 글꼴 프로그램으로 개별적인 소프트웨어로써 기능하는 것이지만, 글꼴 소프트웨어가 GPL을 적용한 소프트웨어를 생성하는 데 직접적인 영향을 끼치는 것이 아닌, 단순한 에셋의 형태로 사용하는 것이기 때문에, 그러한 글꼴 소프트웨어가 GPL 소프트웨어의 소스에서 통합되는 것으로 판단하기 어렵다.

또한 OFL-FAQ에서도 마찬가지로 글꼴 프로그램이 소프트웨어에서 의존적인 개념이 아닌 단순 집합의 개념이기 때문에, 다른 라이선스의 비호환성에 대해 크게 우려하지 않아도 된다고 설명합니다. 따라서 Pretendard의 소스 코드를 직접 빌드해 사용하는 것이 아닌, 배포된 글꼴 파일을 에셋의 형태로 GPL 소프트웨어에 적용하는 방법에는 문제가 되지 않아 보이며, 질문 주신 것과 같이 Pretendard를 번들로 제공하는 형태로는 더더욱 문제가 없어 보입니다.

그러나 이것은 어디까지나 개인적인 판단에 근거한 해석이므로, 정확한 해석이 필요하시다면 해당 분야 전문가와 상담하시는 것을 추천드립니다. 이어서 GPL처럼 전염성이 있는 라이선스로 배포해야 하는 상황이 아닌 OFL 보존에 문제가 없는 일반적인 상황이라면, 같은 OFL로 배포되는 Inter, 본고딕, 나눔고딕, 배달의민족 글꼴의 사용 예와 같이 Pretendard 또한 자유롭게 사용하실 수 있습니다.

질문에 도움이 되었기를 바랍니다. 감사합니다.

**Comment by tribela** [comment]
임베딩 목적으로는 사용해도 되는 것이군요. OFL이 나오게 된 계기를 보면 그게 맞을 것 같습니다. 감사합니다. 😃

**Comment by orioncactus** [comment]
도움이 될 수 있어 다행입니다. 이제 이 이슈는 종결 처리하도록 하겠습니다! ヾ(•ω• )o
